diff --git a/TMessagesProj/src/main/java/org/telegram/ui/Components/ChatActivityEnterView.java b/TMessagesProj/src/main/java/org/telegram/ui/Components/ChatActivityEnterView.java
index 544e3051..4665a951 100644
--- a/TMessagesProj/src/main/java/org/telegram/ui/Components/ChatActivityEnterView.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/Components/ChatActivityEnterView.java
@@ -14473,8 +14473,6 @@ public class ChatActivityEnterView extends FrameLayout implements
                 && (botButton == null || botButton.getVisibility() != VISIBLE)
                 && (notifyButton == null || notifyButton.getVisibility() != VISIBLE)
                 && (scheduledButton == null || scheduledButton.getVisibility() != VISIBLE)
-                && (giftButton == null || giftButton.getVisibility() != VISIBLE)
-                && (suggestButton == null || suggestButton.getVisibility() != VISIBLE)
                 && (sendButton == null || sendButton.width() <= dp(44))
                 && (slowModeButton == null || slowModeButton.getVisibility() != VISIBLE)
                 && emojiButtonPaddingAlpha > 0.99f && attachLayoutPaddingAlpha > 0.99f);
