diff --git a/TMessagesProj/build.gradle b/TMessagesProj/build.gradle
index 4ec4ab9a..e69f849f 100644
--- a/TMessagesProj/build.gradle
+++ b/TMessagesProj/build.gradle
@@ -243,6 +243,10 @@ android {
             buildConfigField "boolean", "BUNDLE", "false"
             buildConfigField "int", "VERSION_NUM", "7"
         }
+        all {
+            buildConfigField "String", "NEBULA_BUILD_DATE", "\"${project.findProperty('nebulaBuildDate') ?: new Date().format('dd.MM.yyyy', TimeZone.getTimeZone('UTC'))}\""
+            buildConfigField "int", "TELEGRAM_VERSION_CODE", APP_VERSION_CODE.toString()
+        }
     }
 
     namespace 'org.telegram.messenger'
